mattermost-plugin/server/configuration.go

```diff
@@ -71,21 +67,23 @@ func (p *Plugin) setConfiguration(configuration *configuration) {
 }
 
 // OnConfigurationChange is invoked when configuration changes may have been made.
-func (p *Plugin) OnConfigurationChange() error {
+func (p *Plugin) OnConfigurationChange() {
 	// Have we been setup by OnActivate?
 	if p.wsPluginAdapter == nil {
-		return nil
+		return
 	}
 
-	configuration := &configuration{}
-	// Load the public configuration fields from the Mattermost server configuration.
-	if err := p.API.LoadPluginConfiguration(&configuration); err != nil {
-		return errors.Wrap(err, "failed to load plugin configuration")
+	mmconfig := p.API.GetUnsanitizedConfig()
```

Comment:
With this changes you are violating a lot of assumtions made by the plugins system. The `configuration` struct is expected to be defined and used to store and restore the configuration of the plugin itself. Also, the plugin API has methods to set and retrieve that configuration, this changes looks like a bit misleading for me. In the future we can try to use that methods and realized that there are no longer usable. Is there any reason to not leverage the existing `configuration` struct to inform the plugin about the current status of the config, and pass to the server what is needed?

Reply:
I added back the property and setting the `configuration` struct. However, the plugin API `LoadPluginConfiguration` only loads the plugin configuration...FeatureFlags are NOT returned. So the call to `GetConfig` is still required. I did update from `GetUnsanitizedConfig`

Reply:
Yes, I agree with that, we need both inside, but my concern is about how that data is exposed everywhere else in the plugin side.